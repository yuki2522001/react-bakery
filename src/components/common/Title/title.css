.title {
  font-size: 16px;
  font-weight: bold;
  cursor: pointer;
}