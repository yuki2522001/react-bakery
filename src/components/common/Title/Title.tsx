import React, { memo } from "react";
import "./title.css";

interface TitleProps {
  text?: string;
  className?: string;
}

const Title: React.FC<TitleProps> = memo(({ text="", className }) => {
  return (
    <h2 data-testid="title-test" className={className}>
      {text}
    </h2>
  );
});

export default Title;
