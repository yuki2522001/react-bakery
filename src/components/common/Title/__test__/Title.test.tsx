import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import Title from "../Title";

describe("Title component", () => {
  test("should render h2 tag", () => {
    const { getByTestId } = render(<Title text="Cheese pocket" />);
    expect(getByTestId("title-test")).toBeInTheDocument();
  });

  test("matches snapshot", () => {
    const { asFragment } = render(<Title text="Cheese pocket" />);
    expect(asFragment()).toMatchSnapshot();
  });
})