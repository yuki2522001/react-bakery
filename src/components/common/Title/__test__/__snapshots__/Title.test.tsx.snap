// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title component matches snapshot 1`] = `
<DocumentFragment>
  <h2
    data-testid="title-test"
  >
    Cheese pocket
  </h2>
</DocumentFragment>
`;
